pinot-common/src/main/java/org/apache/pinot/pql/parsers/pql2/ast/FunctionCallAstNode.java

```diff
@@ -75,22 +80,53 @@ public String getExpression() {
     return _expression;
   }
 
-  public AggregationInfo buildAggregationInfo() {
+  AggregationInfo buildAggregationInfo() {
     String expression;
     // COUNT aggregation function always works on '*'
-    if (_name.equalsIgnoreCase("count")) {
+    if (_name.equalsIgnoreCase(AggregationFunctionType.COUNT.getName())) {
       expression = "*";
     } else {
       List<? extends AstNode> children = getChildren();
-      if (children == null || children.size() != 1) {
-        throw new Pql2CompilationException("Aggregation function expects exact 1 argument");
+      if (children == null || children.size() < 1) {
+        throw new Pql2CompilationException("Aggregation function expects non-null argument");
+      }
+      if (!_name.equalsIgnoreCase(AggregationFunctionType.DISTINCT.getName())) {
+        if (children.size() != 1) {
+          throw new Pql2CompilationException("Aggregation function expects exactly 1 argument as column name");
+        } else {
+          expression = TransformExpressionTree.getStandardExpression(children.get(0));
+        }
+      } else {
+        // DISTINCT
+        if (!Pql2Compiler.ENABLE_DISTINCT) {
+          throw new Pql2CompilationException("Support for DISTINCT is currently disabled");
+        }
+        if (children.size() == 1) {
+          // single column DISTINCT query
+          // e.g SELECT DISTINCT(col) FROM foo
+          expression = TransformExpressionTree.getStandardExpression(children.get(0));
```

Comment:
Do we want to support `SELECT DISTINCT * FROM table`?

Reply:
Discussed offline. We don't support that.